GetStream/stream-chat-android
# [CAS-1130] Cleanup Channel's and User's extensions

https://stream-io.atlassian.net/browse/CAS-1130

### 🎯 Goal

Simplify getting User's and Channel's `name` and `image`

### 🛠 Implementation details

Added `image` and `name` virtual properties to `Channel` and `User`.
`name` and `image` are still being stored inside the `extraData` but I introduced new properties so we don't have to add an additional import for extensions.
Creating `image` and `name` as real properties would be a breaking change. Consider the following code:
```
        val id = "id"
        val type = "messaging"
        val channel = Channel(id =id, type =type).apply { 
            name = "name"
            image = "image"
        }
        client.createChannel(id, type, channel.extraData)
```
We don't pass the whole `Channel` object when creating or updating the channel so the code snippet above won't be working if we convert those extensions into properties.
Moreover, the same applies if someone is using  `extraData["name"]` directly.

### 🧪 Testing

1. Check if images and names are displayed properly in the sample app
2. Try to create a group channel with a name and see if it works correctly.

### ☑️ Checklist
